Fill.py

```diff
@@ -137,6 +140,10 @@ def distribute_items_restrictive(world: MultiWorld, gftower_trash=False, fill_lo
     world.random.shuffle(fill_locations)
     fill_restrictive(world, world.state, fill_locations, progitempool)
 
+    if nonexcludeditempool:
+        world.random.shuffle(fill_locations)
+        fill_restrictive(world, world.state, fill_locations, nonexcludeditempool)
```

Comment:
logic filling all non excluded items would slow down fill drastically. If there's only excluded items left, we might want a fast-track function that just checks exclusion.